### ISSUE TITLE
Elasticnet inconsistencies

### ISSUE THREAD

**Issue by xloffree** [SelfExplainML/PiML-Toolbox#42]
This is a follow up from issue #26 .

<img width="1074" alt="Screenshot 2023-07-29 at 7 04 05 PM" src="https://github.com/SelfExplainML/PiML-Toolbox/assets/64742457/2bc0d729-de1f-49f2-983c-68942554ea59">
<img width="934" alt="Screenshot 2023-07-29 at 7 04 17 PM" src="https://github.com/SelfExplainML/PiML-Toolbox/assets/64742457/06346e30-80dc-426c-9502-4453af8420b3">
<img width="787" alt="Screenshot 2023-07-29 at 7 04 28 PM" src="https://github.com/SelfExplainML/PiML-Toolbox/assets/64742457/810dba51-1ec0-4b30-a296-523da0c7d366">
<img width="1068" alt="Screenshot 2023-07-29 at 7 04 51 PM" src="https://github.com/SelfExplainML/PiML-Toolbox/assets/64742457/1858036b-fe6d-4276-b69f-8009966b4d84">
<img width="1002" alt="Screenshot 2023-07-29 at 7 06 26 PM" src="https://github.com/SelfExplainML/PiML-Toolbox/assets/64742457/a7f9f0ab-4e08-4ca3-8755-d4b458792e23">

When using GLMRegressor, I get very different results than from when I use the method shown in the README, which gives a very poor R2 and says all coefficients are 0. I am certain that this is wrong because we have run elastic net on this data in R and python and have gotten similar results that show a much better R2. The results from the glmRegressor look more reasonable. I am trying to use piml to compare many models at once on the same train/test split (the elastic net models with specific parameters included). We would like to use the results from PiML in an upcoming paper, but this problem with PiML's elastic net is preventing us from being able to do so. Please let me know if there is a solution to this.

In summary, I want to find a way to run elastic net using the same train/test split as when I run all of the other models in PiML. Is there a way to do this? 

Thank you

**Comment by ZebinYang** [comment]
Hi, in the default piml workflow, all the variables are standardized to be within 0 and 1 before modeling. The results for the elastic net can be very different for data with different scales (even with the same regularization strength).

In the screenshot you provided, a GLMRegressor is fitted on the raw data, so the coefficients look reasonable. However, if you want to get similar results using the piml workflow, you may need to decrease the regularization strengths accordingly.

**Comment by xloffree** [comment]
Ok I see, thank you. So the GLMRegressor is a normal glm function whereas the piml workflow standardizes the data? Therefore, the GLM results from piml workflow and the results from GLMRegressor should not be the same?

Thank you!

**Comment by ZebinYang** [comment]
GLMRegressor is a high-code estimator, and you may use it anywhere just like other machine learning models. 

PiML workflow is the pipeline of data preprocessing + GLMRegressor, so their results are different. 

You can get the same results if you do manual data standardizing and then apply it to GLMRegressor.
